autogen/retrieve_utils.py

```diff
@@ -299,3 +285,71 @@ def query_vector_db(
         where_document={"$contains": search_string} if search_string else None,  # optional filter
     )
     return results
+
+def create_lancedb_from_dir(dir_path, max_tokens, db_path, table_name, chunk_mode,
+                            must_break_at_empty_line, embedding_model_name):
+    """Create a LanceDB from all the files in a given directory."""
+    db = LanceDB.connect(db_path)
+    try:
+        # Load embedding model
+        #embeddings = HuggingFaceEmbeddings(model_name=embedding_model_name, model_kwargs={'device': 'cpu'})
+        embedding_function = ef.SentenceTransformerEmbeddingFunction(embedding_model)
+        # Initialize your embedding function (replace it with your actual embedding module)
+        embedding_function = SentenceTransformerEmbeddings(name=embedding_model_name)
+        
+        table = db.get_table(table_name)
+
+        chunks = split_files_to_chunks(get_files_from_dir(dir_path), max_tokens, chunk_mode, must_break_at_empty_line)
+        print(f"Found {len(chunks)} chunks.")
+
+        for i, chunk in enumerate(chunks):
```

Comment:
Better to insert in batch in case there are too many chunks to insert. Btw, is there a `upsert` method for lancedb?